src/hmem_cuda.c

```diff
@@ -111,7 +114,13 @@ static struct cuda_ops cuda_ops = {
 	.cudaSetDevice = cudaSetDevice,
 	.cudaIpcOpenMemHandle = cudaIpcOpenMemHandle,
 	.cudaIpcGetMemHandle = cudaIpcGetMemHandle,
-	.cudaIpcCloseMemHandle = cudaIpcCloseMemHandle
+	.cudaIpcCloseMemHandle = cudaIpcCloseMemHandle,
+
+#if CUDART_VERSION >= 11030
```

Comment:
I am not sure whether checking this hard-coded version is the best approach. A better implementation could be having a dedicated macro like `HAVE_CUDADEVICEFLUSHGPUDIRECTRDMAWRITES` and use `AC_CHECK_DECL` in `configure.ac` to assign 1. See example in https://github.com/ofiwg/libfabric/blob/main/prov/efa/configure.m4#L93-L96.

But I am not sure whether it is over-engineering. I will let @shefty comment here.

Reply:
Doing an explicit check against the feature is strongly preferred.  Though since we're loading it dynamically, what have the check at all?

Reply:
>Though since we're loading it dynamically

That only works for `--enable-cuda-dlopen`. If user drop this flag, libfabric will compile with `libcudart.so`. If that function is not available in cuda library, we will have a compilation error for `.cudaDeviceFlushGPUDirectRDMAWrites = cudaDeviceFlushGPUDirectRDMAWrites,` , right?

Reply:
No I put a guard around that...
```
#if CUDART_VERSION >= 11030
	.cudaDeviceFlushGPUDirectRDMAWrites = cudaDeviceFlushGPUDirectRDMAWrites,
#else
	.cudaDeviceFlushGPUDirectRDMAWrites = NULL,
#endif
```

Reply:
I am not saying the current PR is wrong. I was answering Sean's question of 

>Though since we're loading it dynamically, what have the check at all?

I do not think we are loading it dynamically in all cases. This check should be in compilation as well. Though I was suggesting a different way to check that.